src/cose/decoder.rs

```diff
@@ -144,19 +102,68 @@ pub fn decode_signature(bytes: Vec<u8>, payload: &[u8]) -> Result<Vec<CoseSignat
     let signature_bytes = &cose_signature[2];
     let signature_bytes = unpack!(Bytes, signature_bytes).clone();
     let sig_structure_bytes = get_sig_struct_bytes(
-        cose_sign_array[0].clone(), // Protected body header.
+        protected_body_head,
         protected_signature_header_serialized.clone(),
         payload,
     );
 
-    let signature = CoseSignature {
+    Ok(CoseSignature {
         signature_type: signature_algorithm,
         signature: signature_bytes,
         signer_cert: ee_cert,
         certs: Vec::new(),
```

Comment:
Do we want to grab these out of the protected_body_head here?

Reply:
We should :D That's why I want to move on #10 so that we actually use all the parts and don't mess up things like this.